internal/configs/resource.go

```diff
@@ -699,7 +715,7 @@ type ProviderConfigRefMapping struct {
 
 // HasAlias returns true if the provider is referenced by alias.
 func (m *ProviderConfigRefMapping) HasAlias() bool {
-	return len(m.Aliases) != 0
+	return m == nil || len(m.Aliases) != 0
```

Comment:
Why would HasAlias return true if m == nil?

Reply:
That's a typo, thanks! [302cdc0](https://github.com/opentofu/opentofu/pull/1960/commits/302cdc036887666de1113dace20e45d50f3dfc24)

Reply:
In what scenario is m nil?